### DISCUSSION TITLE
Controls in Home app

### DISCUSSION THREAD

**Discussion by geolocura** [csirikak/homebridge-molekule#15] [General]
Good evening,
I just installed the Molekule plugin on Homebridge (I wish I had come across this one sooner!) and I'm trying to undertand the controls in the Home app. In the device settings, the Home app proposed 3 controls: an on/off switch called "Mode", a slider for Fan Speed, and a toggle between Manual and Auto. I get the slider, but I'm not sure what the other 2 controls actually do, or how they relate to the settings in the Molekule native app. Can you please help me understand them? Thanks!

**Comment by csirikak** [comment]
Hi, 
The switch called Mode is the power toggle, in the off position the air purifier doesn't move any air. In the app this is equivalent to setting it to standby. Moving the fan speed slider to 0 has the same effect.

Auto/Manual is a required characteristic for the air purifier accessory so depending on which model you have, it may either do nothing (air mini) or enable auto protect.

**Reply by geolocura** [reply]
Thank you very much for your help! 

The model I have is Molekule Air. The mode switch does indeed turn it off, and the manual/auto toggle doesn't appear to have any function on this model. 

The "slider" does change the speed between "silent", "auto" and "boost" but the operation appears to be somewhat delicate, requiring the speed to be set at very specific levels to match the 3 preset speeds, otherwise the result it a bit unpredictable. Does that make sense?

**Reply by csirikak** [reply]
Would you be willing to share some logs in debug mode? I would like to be able to know what is the internal model name so I can add it to the devices.json file with the correct fan speed scale.

- I'm assuming that it has 3 levels is that right? The default behavior assumes that a device has 6 speeds.
- Is the "auto" mode truly a dynamic mode that changes depending on the air quality?
- Finally, does it support air quality reporting?

**Reply by geolocura** [reply]
I can definitely do that! I'll take a look at how to extract the logs tomorrow. 

I'm assuming that it has 3 levels is that right?

That is correct. They are called "silent", "auto" and "boost" but they are just fancy names for preset speeds. !

Here's a screenshot of the 3 modes: [image](https://github.com/user-attachments/assets/9957da3b-b7c8-43fb-bbba-f98065c41d3e)

Is the "auto" mode truly a dynamic mode that changes depending on the air quality?

On this model, it's just a name for a medium speed. This model doesn't have any sensing capacity. (I always found the "auto" designation to be misleading actually :-) )

Finally, does it support air quality reporting?

No it doesn't.

**Reply by geolocura** [reply]
Good evening, 
As you suggested, the Molekules I have are an unknown model to the plugin: 

[9/30/2024, 8:00:54 PM] [homebridge-molekule] ✓ Valid Login Credentials
[9/30/2024, 8:00:54 PM] [homebridge-molekule] Adding new accessory: Upstairs Molekule
[9/30/2024, 8:00:54 PM] [homebridge-molekule] The device Upstairs Molekule is not a known model. Using default values.
[Upstairs Molekule@@Filter Change Indication] Characteristic not in required or optional characteristic section for service AirPurifier. Adding anyway.
[Upstairs Molekule@@Filter Life Level] Characteristic not in required or optional characteristic section for service AirPurifier. Adding anyway.
[9/30/2024, 8:00:54 PM] [homebridge-molekule] Adding new accessory: Livingroom Molekule
[9/30/2024, 8:00:54 PM] [homebridge-molekule] The device Livingroom Molekule is not a known model. Using default values.
[Livingroom Molekule@@Filter Change Indication] Characteristic not in required or optional characteristic section for service AirPurifier. Adding anyway.

I may need some guidance to extract the debug log you would need to troubleshoot further.

**Reply by geolocura** [reply]
I cannot find a reference to the internal model name in the log. What I could determine, however, is the relation between the speeds:

Silent: "fanSpeed":1
Auto: "fanSpeed":2
Boost: "fanSpeed":3

Any other speed (4, 5) creates an unpredictable behaviour.

Setting to speed 0 returns the following: [10/3/2024, 6:38:13 PM] [homebridge-molekule] Attempted to set: 0 state on device: Upstairs Molekule Server Reply: {}

**Reply by geolocura** [reply]
And I think I found the log entry you need: 

10/3/2024, 6:43:02 PM] [homebridge-molekule] {"content":[{"serialNumber":"MH1M-LHA180626-001554","name":"Upstairs Molekule","subProduct":{"id":3,"name":"Molekule Air","product":{"id":3,"name":"MH","description":"Molekule Air","serialNumberPrefix":"MH","health":["sn","fw","dc"]},"createdBy":"admin-app@molekule.com","createdAt":"2020-03-18T12:11:58.462Z","updatedBy":"admin-app@molekule.com","updatedAt":"2020-03-18T12:11:58.462Z","serialNumberPrefix":"MH1"},"firmwareVersion":"FW:8.31.2","model":"Molekule Air","firstConnectedUser":"c95650c2-43c0-4bb6-b915-48533c412d1e","firstConnectionDate":"2020-03-19T12:09:08.064Z","macAddress":"E0:4F:43:A8:E1:B9","pecoFilter":"0","fanspeed":"2","online":"true","preFilter":"0","silent":"","burst":"N/A"},{"serialNumber":"MH1M-PEA190618-003997","name":"Livingroom Molekule","subProduct":{"id":3,"name":"Molekule Air","product":{"id":3,"name":"MH","description":"Molekule Air","serialNumberPrefix":"MH","health":["sn","fw","dc"]},"createdBy":"admin-app@molekule.com","createdAt":"2020-03-18T12:11:58.462Z","updatedBy":"admin-app@molekule.com","updatedAt":"2020-03-18T12:11:58.462Z","serialNumberPrefix":"MH1"},"firmwareVersion":"FW:8.31.2","model":"Molekule Air","firstConnectedUser":"c95650c2-43c0-4bb6-b915-48533c412d1e","firstConnectionDate":"2020-03-12T11:00:48.140Z","macAddress":"3C:E1:A1:E2:4D:58","pecoFilter":"0","fanspeed":"2","online":"true","preFilter":"0","silent":"","burst":"N/A"}],"last":true,"totalElements":2,"totalPages":1,"first":true,"number":0,"numberOfElements":2,"size":10}

**Reply by csirikak** [reply]
Thanks! Could you try version [1.4.4-alpha](https://www.npmjs.com/package/homebridge-molekule/v/1.4.4-alpha) and see if it works with your purifier?

**Reply by geolocura** [reply]
Thank you! I have installed it this morning and at the first sight, it appears that I can now use the full scope of the slider. I will monitor over the weekend and I will let you know how it behaves.

**Reply by geolocura** [reply]
Good evening! I just want to confirm that version 1.4.4 alpha seems to play well with Molekule Air. Thanks a lot!
